packages/core/src/components/Tabs/Tabs.jsx

```diff
@@ -134,6 +175,75 @@ export class Tabs extends React.PureComponent {
     }
   }
 
+  /**
+   * Build tabs array to handle arrow keypress events accurately
+   */
+  tabsArrayLen() {
+    return React.Children.toArray(this.props.children)
+      .filter(isTab);
+  }
+
+  /**
+   * Find the current tab index on left or right arrow keypress
+   * @param {Array} React Tab components
+   * @param {String} ID prop of currently selected Tab
+   * @return {Number} Returns index of currently selected array item
+   */
+  tabFindIndex(tabsArr, findId) {
+    const index = tabsArr.findIndex(elem => elem.props.id === findId);
+
+    return index;
+  }
+
+  /**
+   * Handle the left arrow keydown events
+   * @param {Object} Native event object, used to listen for left arrow keycode
+   * @param {String} ID prop of currently selected Tab
+   * 
+   * The index returned from tabFindIndex is used in a reduce by 1 logic
+   * tree to ensure the right tab[index] ID is targeted for updated state.
+   * 
+   * The replaceState function did not like the href argument passed into 
+   * handleTabClick, so passing an interpolated hashbang plus selected Tab
+   * string instead. 
+   */
+  tabSwitchLeft(evt, curId) {
```

Comment:
We should try using consistent argument names here. It keeps things easier to follow, but also helps catch things like: The id being passed into this method right now is `panelId` — based on the description of this argument though, you might be expecting the tab id instead?